dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/programrule/implementers/AbstractRuleActionImplementer.java

```diff
@@ -211,19 +214,43 @@ public Map<String, List<EnrollmentActionRule>> getEnrollmentEffects(
     }
 
     private List<Attribute> mergeAttributes(
-        List<Attribute> enrollmentAttributes, List<Attribute> attributes )
+        TrackerPreheat preheat, List<Attribute> enrollmentAttributes, List<Attribute> attributes )
     {
 
-        List<String> payloadAttributes = attributes.stream()
+        // TODO(DHIS2-12563) the merged attributes are passed to
```

Comment:
@enricocolasante, @stian-sandvold or @zubaira when migrating the tracker.domain.Attribute.attribute from String to MetadataIdentifier (so idScheme aware type) I stumbled on a few program rule (engine) related pieces of code.

Every time we deal with metadata from user input it can be in any idScheme. Rule engine has no concept of idSchemes, assumes identifiers are UIDs.

So here for I need to map the payload attribute identifiers to UIDs.

Do you see any risks associated with this here?

Reply:
I think here is the right place to convert all idSchemed identifiers to UID